enrichment-mongo/src/main/java/eu/europeana/enrichment/mongo/dao/TopicEntityDaoImpl.java

```diff
@@ -0,0 +1,99 @@
+package eu.europeana.enrichment.mongo.dao;
+
+import static dev.morphia.query.experimental.filters.Filters.eq;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import dev.morphia.Datastore;
+import dev.morphia.query.internal.MorphiaCursor;
+import eu.europeana.enrichment.common.commons.AppConfigConstants;
+import eu.europeana.enrichment.model.TopicEntity;
+import eu.europeana.enrichment.model.TopicModel;
+import eu.europeana.enrichment.model.impl.TopicEntityImpl;
+import eu.europeana.enrichment.model.impl.TopicModelImpl;
+
+@Repository(AppConfigConstants.BEAN_ENRICHMENT_TOPIC_ENTITY_DAO)
+public class TopicEntityDaoImpl implements TopicEntityDao{
+
+	
+
+	@Autowired
+	private Datastore enrichmentDatastore;
+	
+	@Override
+	public void save(TopicEntity topicEntity) {
```

Comment:
Please see the new save method in the other entities like StoryEntityImpl or ItemEntityImpl in the develop branch, and change this code accordingly. Important is, before calling the save method to make sure, the created entity is the one from the db, if it exists there.

Reply:
I think we can merge first and them I can update the code. I cannot explore the new implementation now